### PR TITLE
Initial layout

### PR DESCRIPTION
**Pull request by Phosphorus-M**
Agregado en la landing page:
- Header
- Hero
- Acerca
- Nuestros proyectos

En teoría todo es responsive de momento, faltan varias cosas pero bueno, es un comienzo. 

El diseño que se viene usando es [este](https://github.com/rust-lang-ar/rust-lang-ar.github.io/discussions/8#discussioncomment-765321) desde luego esta abierto a cambios y sugerencias.

### CODE REVIEW

**Comment by EstebanBorai** [comment]
@Phosphorus-M I think it would be nice to have screenshots here, so we can see what's being merged

**Comment by EstebanBorai** [comment]
@Phosphorus-M please fix GitHub actions `clippy` and `fmt` issues, its important to keep our codebase as clean as possible, remember to run both commands before pushing to GitHub.

```sh
cargo fmt && cargo clippy
```

File: src/components/our_projects.rs [hunk 1]
```diff
@@ -0,0 +1,167 @@
+use yew::{html, Component, ComponentLink, Html, ShouldRender};
+use yew::services::fetch::{FetchService, FetchTask, Request, Response};
+use super::{project_card::{ProjectCard}};
```
Comment (EstebanBorai) [inline src/components/our_projects.rs:3]: The inner curly braces are not required.

```suggestion
use super::project_card::ProjectCard};
```

File: src/components/our_projects.rs [hunk 2]
```diff
@@ -0,0 +1,167 @@
+use yew::{html, Component, ComponentLink, Html, ShouldRender};
```
Comment (EstebanBorai) [inline src/components/our_projects.rs:1]: I suggest sorting imports alphabetically and also by its kind.

I usually do:

```rust
// Introduce local modules to context
mod some_module;

// Import dependency crates (external)
use anyhow::{bail, Error};
use yew;

// Import crate (but not sibling) (think of this as crate-absolute path, similar to Python's)
use crate::components::MyBeautifulComponent;

// Module relative paths
use self::some_sibling_module::Sibling;
```

File: src/components/our_projects.rs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+use anyhow::Error;
+use serde::Deserialize;
+
+const GITHUB_REPOS_URL: &str = "https://api.github.com/orgs/rust-lang-ar/repos";
+
+  
+pub struct OurProjects {
+    is_fetching: bool,
+    fetch_failed: bool,
+    _fetch_task: Option<FetchTask>,
```
Comment (EstebanBorai) [inline src/components/our_projects.rs:15]: Why do we have underscore-prefixed field names? If we have field access keywords for that?

File: src/components/project_card.rs [hunk 4]
```diff
@@ -0,0 +1,108 @@
+use yew::{html, Component, ComponentLink, Html, Properties, ShouldRender};
+
+#[derive(Clone, Copy,PartialEq, Eq, Hash)]
+pub enum Lang {
+    Rust,
+    CSS,
```
Comment (EstebanBorai) [inline src/components/project_card.rs:6]: Please remember to run `cargo clippy`, Rust doesn't make use of all caps identifiers

File: src/components/project_card.rs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    fn change(&mut self, _: Self::Properties) -> ShouldRender {
+        true
+    }
+
+    fn rendered(&mut self, _first_render: bool) {
+
+    }
+
+    fn view(&self) -> Html {
+        let (_class_name, language_text) = self.language.markup_resources();
```
Comment (EstebanBorai) [inline src/components/project_card.rs:91]: If you are not using the first member of the tuple, you can just do:

```suggestion
        let (_, language_text) = self.language.markup_resources();
```

But as there's only two we are writing so much here, you can have it even more simple if you just do classic assign here:

```suggestion
        let language_text = self.language.markup_resources().1;
```

File: src/img/fa-brands_meetup.svg [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+<feGaussianBlur stdDeviation="1.5"/>
+<feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"/>
+<feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow"/>
+<feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow" result="shape"/>
+</filter>
+<clipPath id="clip0">
+<rect width="70" height="70" fill="white" transform="translate(3 2)"/>
+</clipPath>
+</defs>
+</svg>
```
Comment (EstebanBorai) [inline src/img/fa-brands_meetup.svg:20]: I suggest optimizing SVG using a tool like SVGO:

https://github.com/svg/svgo

File: static/fa-brands_telegram.svg [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<svg width="62" height="64" viewBox="0 0 62 64" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (EstebanBorai) [inline static/fa-brands_telegram.svg:1]: I suggest optimizing this SVG using a tool like https://github.com/svg/svgo

File: tailwind.config.js [hunk 8]
```diff
@@ -1,11 +1,35 @@
...
+      'project-card': "420px"
+    },
+    extend: {
+      colors:{
+        header: "#E8C64E"
+      },
+      fontFamily:{
+        playfair: ["Playfair Display"],
+        lato: ["Lato"],
+      },
```
Comment (EstebanBorai) [inline tailwind.config.js:24]: I suggest using semantic names for this, instead of using the font names itself.
Also, don't forget the font fallback.

```suggestion
      fontFamily:{
        heading: ['Playfair Display', 'sans-serif'],
        body: ['Lato', 'sans-serif'],
      },
```

File: tailwind.config.js [hunk 9]
```diff
@@ -1,11 +1,35 @@
...
   theme: {
-    extend: {},
+    borderColor: theme => ({
+      ...theme('colors'),
+      'rust': "#e56034"
+    }),
+    minWidth: {
+      'project-card-mobile': "320px",
+      'project-card': "420px"
+    },
```
Comment (EstebanBorai) [inline tailwind.config.js:16]: Doesn't this belongs to the CSS itself? We have media queries and also the `@screen` directive.
Check this out: https://tailwindcss.com/docs/breakpoints and https://tailwindcss.com/docs/functions-and-directives#screen

**Review by EstebanBorai — CHANGES_REQUESTED** [review]

File: src/components/our_projects.rs (hunk 2 above)
Comment (Phosphorus-M) [reply src/components/our_projects.rs:1]: Oh nice, it's a good suggestion, I'll change the code!

File: tailwind.config.js (hunk 8 above)
Comment (Phosphorus-M) [reply tailwind.config.js:24]: In this case, I do not agree because the problem is that Lato not only is used in the body, in some cases is used to be the font of titles, and in other cases, Playfair is used in subtitles. 
It's complicated to use a semantic name to the fonts.

Comment (EstebanBorai) [reply tailwind.config.js:24]: It doesn't matter, actually design is more dynamic than it seems. The goal is to have a semantic name but only place I've seen some strictness on usage is in a Word document 😂 .

Im just suggesting to use semantic names over the name of the font-family as Tailwind is intended for standardization, instead of having an arbitrary set of font names all around the code you stick to "aliases" to these names.

File: src/router.rs [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+#[derive(Clone, Debug, Switch)]
+pub enum AppRoute {
+    #[to = "/resources"]
+    Resources,
+    #[to = "/events"]
+    Events,
+    #[to = "/about"]
+    About,
+    #[to = "/!"]
```
Comment (EstebanBorai) [inline src/router.rs:12]: Is this exclamation mark required here?

File: src/router.rs [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+    pub fn into_route(self) -> Route {
+        Route::from(self.into_public())
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct PublicUrlSwitch(AppRoute);
+impl PublicUrlSwitch {
+    fn base_url() -> Url {
```
Comment (EstebanBorai) [inline src/router.rs:28]: Can we have a comment on why is this required?

File: src/router.rs [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+        if let Ok(Some(href)) = yew::utils::document().base_uri() {
+            Url::new(&href).unwrap()
+        } else {
+            Url::new("/").unwrap()
+        }
+    }
+
+    fn base_path() -> String {
```
Comment (EstebanBorai) [inline src/router.rs:36]: Please can you elaborate a bit on why is this required?

File: src/router.rs [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+        PublicUrlSwitch(self)
+    }
+
+    pub fn into_route(self) -> Route {
+        Route::from(self.into_public())
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct PublicUrlSwitch(AppRoute);
```
Comment (EstebanBorai) [inline src/router.rs:26]: Why do we need a Public URL Switch?

File: static/images/arrow-down-solid.svg [hunk 14]
```diff
@@ -0,0 +1 @@
+<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="arrow-down" class="svg-inline--fa fa-arrow-down fa-w-14" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"></path></svg>
```
Comment (EstebanBorai) [inline static/images/arrow-down-solid.svg:1]: Are these `fa` classes required?

File: static/images/fa-brands_telegram.svg [hunk 15]
```diff
@@ -0,0 +1 @@
+<svg width="62" height="64" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill="#fff" d="M12 13h36v38H12z"/><path d="M31 1C13.875 1 0 14.875 0 32s13.875 31 31 31 31-13.875 31-31S48.125 1 31 1zm15.225 21.238-5.087 23.974c-.375 1.7-1.388 2.113-2.8 1.313l-7.75-5.712-3.738 3.6c-.413.412-.763.762-1.563.762l.55-7.887L40.2 25.313c.625-.55-.138-.863-.963-.313l-17.75 11.175-7.65-2.387c-1.662-.525-1.7-1.663.35-2.463L44.076 19.8c1.387-.5 2.6.337 2.15 2.438z" fill="#1D98DC"/></svg>
```
Comment (EstebanBorai) [inline static/images/fa-brands_telegram.svg:1]: I suggest removing `fill`, `height` and `width` attributes from this SVG, and specify `currentColor` for `fill` and use CSS to specify sizes. WDYT?

File: src/styles/tailwind.css [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+@tailwind components;
+@tailwind utilities;
+@tailwind aspect-ratio;
+
+@import url('https://fonts.googleapis.com/css2?family=Playfair+Display&display=swap');
+@import url('https://fonts.googleapis.com/css2?family=Lato&display=swap');
+@import url('https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;700&display=swap');
+
+.bg-general{
+    background: url('https://i.imgur.com/tLt6Zo6.png'), #F4A55D;
```
Comment (EstebanBorai) [inline src/styles/tailwind.css:21]: Why is this resource external while others are being served locally? I would rather having all resources served in a single fashion.

File: webpack.config.js [hunk 17]
```diff
@@ -0,0 +1,65 @@
+const path = require('path');
```
Comment (EstebanBorai) [inline webpack.config.js:1]: I think webpack is too much for this, why not using PostCSS along with Rollup? Is faster and lightweight.

File: webpack.config.js [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+const WasmPackPlugin = require('@wasm-tool/wasm-pack-plugin');
```
Comment (EstebanBorai) [inline webpack.config.js:2]: Is this required? We are using Trunk already so Im not sure why we have a WASM loader here.

**Review by EstebanBorai — CHANGES_REQUESTED** [review]

**Comment by EstebanBorai** [comment]
@Phosphorus-M please make sure you rebase! It makes easier PR reviewing.

**Comment by EstebanBorai** [comment]
The router looks good so far, but I believe we will encounter issues with client side routing given that we are using GitHub Pages which (AFAIK) it doesn't support client-side routing.

We would have to either use a hash router for this or change our hosting provider to make it work as expected.

File: src/router.rs (hunk 10 above)
Comment (Phosphorus-M) [reply src/router.rs:12]: Nop, removed

File: src/router.rs (hunk 13 above)
Comment (Phosphorus-M) [reply src/router.rs:26]: We don't need the struct PublicUrlSwitch, I made a refactoring.
